UI/Theme.cpp

```diff
@@ -0,0 +1,238 @@
+		}
+
+		for (size_t f = 0; f < fileInfo.size(); f++) {
+			IniFile ini;
+			bool success = false;
+			Path name = fileInfo[f].fullName;
+			Path path = directories[d];
+			// Hack around Android VFS path bug. really need to redesign this.
+			if (name.ToString().substr(0, 7) == "assets/")
+				name = Path(name.ToString().substr(7));
+			if (path.ToString().substr(0, 7) == "assets/")
+				path = Path(path.ToString().substr(7));
+
+			if (ini.LoadFromVFS(name.ToString()) || ini.Load(fileInfo[f].fullName)) {
+				success = true;
+				// vsh load. meh.
+			}
+
+			if (!success)
+				continue;
+
+			// Alright, let's loop through the sections and see if any is a themes.
+			for (size_t i = 0; i < ini.Sections().size(); i++) {
+				Section &section = ini.Sections()[i];
+				themeInfo info;
+				section.Get("Name", &info.name, section.name().c_str());
+
+				section.Get("ItemStyleFg", &info.uItemStyleFg, 0xFFFFFFFF);
+				section.Get("ItemStyleBg", &info.uItemStyleBg, 0x55000000);
+				section.Get("ItemFocusedStyleFg", &info.uItemFocusedStyleFg, 0xFFFFFFFF);
+				section.Get("ItemFocusedStyleBg", &info.uItemFocusedStyleBg, 0xFFEDC24C);
+				section.Get("ItemDownStyleFg", &info.uItemDownStyleFg, 0xFFFFFFFF);
+				section.Get("ItemDownStyleBg", &info.uItemDownStyleBg, 0xFFBD9939);
+				section.Get("ItemDisabledStyleFg", &info.uItemDisabledStyleFg, 0x80EEEEEE);
+				section.Get("ItemDisabledStyleBg", &info.uItemDisabledStyleBg, 0x55E0D4AF);
+				section.Get("ItemHighlightedStyleFg", &info.uItemHighlightedStyleFg, 0xFFFFFFFF);
+				section.Get("ItemHighlightedStyleBg", &info.uItemHighlightedStyleBg, 0x55BDBB39);
+
+				section.Get("ButtonStyleFg", &info.uButtonStyleFg, 0xFFFFFFFF);
+				section.Get("ButtonStyleBg", &info.uButtonStyleBg, 0x55000000);
+				section.Get("ButtonFocusedStyleFg", &info.uButtonFocusedStyleFg, 0xFFFFFFFF);
+				section.Get("ButtonFocusedStyleBg", &info.uButtonFocusedStyleBg, 0xFFBD9939);
+				section.Get("ButtonDownStyleFg", &info.uButtonDownStyleFg, 0xFFFFFFFF);
+				section.Get("ButtonDownStyleBg", &info.uButtonDownStyleBg, 0xFFBD9939);
+				section.Get("ButtonDisabledStyleFg", &info.uButtonDisabledStyleFg, 0x80EEEEEE);
+				section.Get("ButtonDisabledStyleBg", &info.uButtonDisabledStyleBg, 0x55E0D4AF);
+				section.Get("ButtonHighlightedStyleFg", &info.uButtonHighlightedStyleFg, 0xFFFFFFFF);
+				section.Get("ButtonHighlightedStyleBg", &info.uButtonHighlightedStyleBg, 0x55BDBB39);
+
+				section.Get("HeaderStyleFg", &info.uHeaderStyleFg, 0xFFFFFFFF);
+				section.Get("InfoStyleFg", &info.uInfoStyleFg, 0xFFFFFFFF);
+				section.Get("InfoStyleBg", &info.uInfoStyleBg, 0x00000000);
+				section.Get("PopupTitleStyleFg", &info.uPopupTitleStyleFg, 0xFFE3BE59);
+				section.Get("PopupStyleFg", &info.uPopupStyleFg, 0xFFFFFFFF);
+				section.Get("PopupStyleBg", &info.uPopupStyleBg, 0xFF303030);
+
+				appendTheme(info);
+			}
+		}
+	}
+}
+
+static UI::Style MakeStyle (uint32_t fg, uint32_t bg) {
```

Comment:
nit: We usually don't put a space before the `(` in arguments.

-[Unknown]

Reply:
PPSSPP